packages/sanity/src/core/store/bundles/createBundlesStore.ts

```diff
@@ -0,0 +1,234 @@
+    ),
+  )
+
+  const listener$ = client.observable.listen<BundleDocument>(QUERY, {}, LISTEN_OPTIONS).pipe(
+    map((event) => ({event})),
+    catchError((error) =>
+      of<ActionWrapper>({
+        action: {
+          type: 'LOADING_STATE_CHANGED',
+          payload: {
+            loading: false,
+            error,
+          },
+        },
+      }),
+    ),
+    // Skip the first event received upon subscription. This `welcome` event would cause the list
+    // to be fetched again, which is not desirable immediately after the initial fetch has occurred.
+    skip(1),
+    // Ignore events emitted while the list fetch is pending.
+    filter(() => !fetchPending$.value),
+    switchMap<ActionWrapper | EventWrapper, Observable<bundlesReducerAction | undefined>>(
+      (entry) => {
+        if ('action' in entry) {
+          return of(entry.action)
+        }
+
+        const {event} = entry
+
+        // After successful reconnection, fetch the list. Note that the first event is skipped, so
+        // this will not occur upon initial connection.
+        if (event.type === 'welcome') {
+          return listFetch$.pipe(delay(1_000))
+        }
+
+        // The reconnect event means that we are trying to reconnect to the realtime listener.
+        // In this case we set loading to true to indicate that we're trying to
+        // reconnect. Once a connection has been established, the welcome event
+        // will be received and we'll fetch all bundles again (above)
+        if (event.type === 'reconnect') {
+          return of<bundlesReducerAction>({
+            type: 'LOADING_STATE_CHANGED',
+            payload: {
+              loading: true,
+              error: undefined,
+            },
+          })
+        }
+
+        // Handle mutations (create, update, delete) from the realtime listener
+        // and update the bundles store accordingly
+        if (event.type === 'mutation') {
+          if (event.transition === 'disappear') {
+            return of<bundlesReducerAction>({type: 'BUNDLE_DELETED', id: event.documentId})
+          }
+
+          if (event.transition === 'appear') {
+            const nextBundle = event.result
+
+            if (nextBundle) {
+              return of<bundlesReducerAction>({type: 'BUNDLE_RECEIVED', payload: nextBundle})
+            }
+
+            return of(undefined)
+          }
+
+          if (event.transition === 'update') {
+            const updatedBundle = event.result
+
+            if (updatedBundle) {
+              return of<bundlesReducerAction>({type: 'BUNDLE_UPDATED', payload: updatedBundle})
+            }
+          }
+        }
+
+        return of(undefined)
+      },
+    ),
+  )
+
+  const state$ = merge(listFetch$, listener$, dispatch$).pipe(
+    filter((action): action is bundlesReducerAction => typeof action !== 'undefined'),
+    scan((state, action) => bundlesReducer(state, action), INITIAL_STATE),
+    startWith(INITIAL_STATE),
+    shareReplay(1),
```

Comment:
Curious on what the `1` is doing here. Checking the docs https://rxjs.dev/api/index/function/shareReplay takes an object as parameter with the signature [ShareReplayConfig](https://rxjs.dev/api/index/interface/ShareReplayConfig)

Reply:
Great question! The function can be called with a config object, but if you have no other options to provide, you can instead provide a `bufferSize` argument directly.

In this case, a `bufferSize` of 1 means only the most recently emitted state object will be stored and broadcast to new subscribers.

Reply:
TIL: Thanks @juice49 !